osgar/lib/test_quaternion.py

```diff
@@ -42,5 +43,9 @@ def test_from_axis_angle(self):
         self.assertAlmostEqual(expected[2], actual[2], places=6)
         self.assertAlmostEqual(expected[3], actual[3], places=6)
 
+    def test_asin_domain(self):
+        q = (0.02089575225593203, -0.6597655110187111, 0.381406681115162, -0.7726979260818011)
```

Comment:
This quaternion tests only the added normalization but it does not test the part when asin overflows.

Reply:
I am having no luck finding any quaternion that would lead to the singularity :worried:

Reply:
As I said - it is border case like minus zero. Neither I have such case but ... I have seen it in my life like in the cosine theorem ... but ... I will remove it ... (I know what will happen in NASA finals ;) ...

Reply:
If we were to remove all code that does not have tests there would not be much left :wink:

Reply:
I would have just adjusted the name of the test since it tests the builtin normalization and not the asin domain problem.